### PR TITLE
Bugfixes for 2.7.0 which correct errors in downstream package builds

### PR DESCRIPTION
**Pull request by jamesba**
Fixes cloudagnostic build error caused by `end_dump` not being called.

### CODE REVIEW

**Comment by jamesba** [comment]
Have tested locally and cloudagnostic 1f57f04840d32888e0603f4ea4b0467d5d4f8a98 (v0.7.4) has no errors when tested with this version of mediagrains.

File: mediagrains/gsf.py [hunk 1]
```diff
@@ -1514,8 +1514,8 @@ def add_grains(self,
     def dump(self):
         """Dump the whole contents of this encoder to the file in one go,
         replacing anything that's already there."""
-        with self:
-            pass
+        self.start_dump(all_at_once=True)
+        self.end_dump(all_at_once=True)
```
Comment (philipnbbc) [inline mediagrains/gsf.py]: Could be `self.end_dump()` but that because all_at_once is not used in end_dump(). That could be left to a separate PR if you want.

Comment (jamesba) [reply mediagrains/gsf.py]: It's not used in this class's implementation, but it was part of the declared interface before, and hence there's no guarantee that subclasses that override `end_dump` won't use it. Hence I've included it.
